doc/syslog/python-logger-enhancement.md

```diff
@@ -0,0 +1,217 @@
+```python
+log = syslogger.SysLogger('daemon_name', enable_runtime_config=True)
+
+def run():
+    while True:
+        # log instance is cloned from main process, it is not registered to the global
+        # log registry of child process
+        log.log_notice('log something')
+
+p = multiprocessing.Process(target=run)
+p.join()
+```
+
+Right way:
+```python
+log = syslogger.SysLogger('daemon_name', enable_runtime_config=True)
+
+def child():
+    child_log = syslogger.SysLogger('daemon_name', enable_runtime_config=True)
+    while True:
+        # child_log is created in child process, so it will register itself to
+        # the global log registry of child process
+        child_log.log_notice('log something')
+
+p = multiprocessing.Process(target=child)
+p.join()
+```
+
+2. Main process shall forward SIGHUP to child processes. There are many options to achieve this, here is an example:
+
+```python
+log = syslogger.SysLogger('daemon_name', enable_runtime_config=True)
+children = []
+
+def child_signal_handler(signum, frame):
+    syslogger.SysLogger.refresh_config()
+
+def child():
+    signal.signal(signal.SIGHUP, child_signal_handler)
+    child_log = syslogger.SysLogger('daemon_name', enable_runtime_config=True)
+    while True:
+        child_log.log_notice('log something')
+
+def main_signal_handler(signum, frame):
+    global children
+    if signum == signal.SIGHUP:
+        syslogger.SysLogger.refresh_config()
+        for child in children:
+            os.kill(child.pid, signal.SIGHUP)
+
+signal.signal(signal.SIGHUP, main_signal_handler)
+p = multiprocessing.Process(target=child)
+children.append(p)
+p.join()
+```
+
+### SAI API
+
+N/A
+
+### Configuration and management
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A
+
+#### CLI/YANG model Enhancements
+
+Currently, `swssloglevel` is used to update log level at run time. It relies on the thread running in `swsscommon.Logger` to refresh the log level of each logger instance. Since there is no such thread in python logger implementation, a new CLI shall be added to refresh the log level by sending signal to relevant daemon.
+
+> Note: `swssloglevel` cannot send signal to daemons running in other container because it is actually a program installed in swss container.
+
+config syslog level -c <component> -l <log_level> [--service <service_name>] [--program <program_name>] [--pid <pid>]
+
+- "-c": component name or log identifier name. It is the key of LOGGER table.
+- "-l": log level. One of "DEBUG", "INFO", "NOTICE", "WARN", "ERROR"
+- "--service": container name. Optional. The signal will be sent to that container if present.
+- "--program": program name. Optional, only applicable when --service is present. The signal will be sent to that program running in the service.
+- "--pid": process ID. Optional. If --service is present, the signal will be sent to the PID running in the service; otherwise, the signal will be send to the PID running in host side.
```

Comment:
@Junchao-Mellanox if user does not specify the optional argument --service, then also this will work?

Reply:
If `--service` is not present, the command will try to send the signal to host side process which has the pid equals to `--pid`